src/main/php/PHPMD/Rule/UnusedFormalParameter.php

```diff
@@ -98,7 +98,21 @@ protected function isInheritedSignature(AbstractNode $node)
         if ($node instanceof MethodNode) {
             $comment = $node->getDocComment();
 
-            return $comment && preg_match('/@inheritdoc/i', $comment);
+            if ($comment && preg_match('/@inheritdoc/i', $comment)) {
+                return true;
+            }
+
+            if (\PHP_VERSION_ID >= 80300 && class_exists($node->getParentType()->getFullQualifiedName())) {
+                // Remove the "()" at the end of method's name.
+                $methodName = substr($node->getFullQualifiedName(), 0, -2);
+                $reflectionMethod = new \ReflectionMethod($methodName);
+
+                foreach ($reflectionMethod->getAttributes() as $reflectionAttribute) {
+                    if ($reflectionAttribute->getName() === 'Override') {
+                        return true;
+                    }
+                }
+            }
```

Comment:
You can return early to get less nesting in your ode 😃 

```suggestion
            if (\PHP_VERSION_ID < 80300 || !class_exists($node->getParentType()->getFullQualifiedName())) {
                return false;
            }
            // Remove the "()" at the end of method's name.
            $methodName = substr($node->getFullQualifiedName(), 0, -2);
            $reflectionMethod = new \ReflectionMethod($methodName);

             foreach ($reflectionMethod->getAttributes() as $reflectionAttribute) {
                if ($reflectionAttribute->getName() === 'Override') {
                    return true;
                }
            }
```

Reply:
Done. Thanks.